docs/machineregistration-reference.md

```diff
@@ -104,22 +107,72 @@ The only required value for a successful installation is the `device` key as we
   ```
 </details>
 
+#### config.elemental.install.device-selector
+
+The `device-selector` field can be used to dynamically pick device during installation. The field contains a list of rules that looks like the following:
+
+<details>
+<summary>Example device-selector based on device name</summary>
+  ```yaml showLineNumbers
+  device-selector:
+  - key: Name
+    operator: In
+    values:
+    - /dev/sda
+    - /dev/vda
+    - /dev/nvme0
+  ```
+</details>
+
+<details>
+<summary>Example device-selector based on device size</summary>
+  ```yaml showLineNumbers
+  device-selector:
+  - key: Size
+    operator: Lt
+    values:
+    - 100Gi
+  - key: Size
+    operator: Gt
+    values:
+    - 30Gi
+  ```
+</details>
+
+The currently supported operators are:
+
+| Operator            | Description                                       |
+| ------------------- | ------------------------------------------------- |
+| In                  | The key matches one of the provided values        |
+| NotIn               | The key does not match any of the provided values |
+| Gt                  | The key is greater than a single provided value   |
```

Comment:
Do we have checks in place to ensure 'value' is numeric ?
(iow - there's plenty of opportunity to break^wtest this 😆 )

Reply:
It's an optional text-field for the operator so it will only be parsed when registering the node, the error message looks like this when running `journalctl -f -u elemental-register-install`

```
...
Nov 28 09:41:33 rancher-22697 elemental-register[1942]: Error: installing elemental: failed picking installation device: failed to parse quantity sda
...
```